hack/generator/pkg/astmodel/identifier_factory.go

```diff
@@ -39,21 +46,37 @@ func NewIdentifierFactory() IdentifierFactory {
 }
 
 // CreateIdentifier returns a valid Go public identifier
-func (factory *identifierFactory) CreateIdentifier(name string) string {
+func (factory *identifierFactory) CreateIdentifier(name string, visibility Visibility) string {
 	if identifier, ok := factory.renames[name]; ok {
-		return identifier
+		name = identifier
 	}
 
 	// replace with spaces so titlecasing works nicely
 	clean := filterRegex.ReplaceAllLiteralString(name, " ")
 
-	titled := strings.Title(clean)
-	result := strings.ReplaceAll(titled, " ", "")
+	result := strings.Title(clean)
+
+	if visibility == Internal {
+		// TODO: This is a hack, as there are cases (acronyms, etc) where
+		// TODO: this doesn't work...
+		// Lowercase the first rune if visibility is internal
+		done := false
+		result = strings.Map(
+			func(r rune) rune {
+				if !done {
+					done = true
+					return unicode.ToLower(r)
+				}
+				return r
```

Comment:
nit: minor: personal preferance I guess but I prefer early exit:
```suggestion
if done {
    return r
}
done = true
return unicode.ToLower(r)
```